libs/test/SmartStoreTest/src/com/salesforce/androidsdk/smartstore/store/SmartStoreTest.java

```diff
@@ -1335,6 +1335,12 @@ public void testGetDatabaseSize() throws JSONException {
 			JSONObject soupElt = new JSONObject("{'key':'abcd" + i + "', 'value':'va" + i + "', 'otherValue':'ova" + i + "'}");
 			store.create(TEST_SOUP, soupElt);
 		}
+
+		// With WAL enabled we must force a WAL checkpoint if we want the actual DB file to reflect the new content:
+		store.getDatabase()
+				.query("PRAGMA wal_checkpoint(FULL);")
+				.moveToNext();
```

Comment:
I felt that this was the most non-invasive way to fix the test, but this does expose some raw DB queries and knowledge about how SQLite works.  The alternative to getting this to pass is to perform enough writes to trigger a checkpoint, but that threshold is determined in the config stage and this test may not be able to know what that page threshold is set to.  This could lead to inconsistent test runs which is why I elected to force the checkpoint regardless of the page threshold.